common/src/main/java/com/strategyobject/substrateclient/common/inject/Injection.java

```diff
@@ -0,0 +1,35 @@
+package com.strategyobject.substrateclient.common.inject;
+
+import com.google.common.base.Preconditions;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Getter
+@RequiredArgsConstructor
+public class Injection<T, S extends Injection<T, S>> {
+    private final @NonNull T receiver;
+
+    private List<Injection<T, ?>> injections;
+
+    @SuppressWarnings("unchecked")
+    public S inject(@NonNull Object... injections) {
+        Preconditions.checkArgument(injections.length > 0);
+
+        this.injections = Arrays.stream(injections)
```

Comment:
I suppose the behaviour can be non-obvious as so it follows the principle of method chaining, but mutates the current state.

Reply:
It mimics the behavior of `ScaleReader.inject()`